kpi/serializers/v2/asset.py

```diff
@@ -343,6 +343,41 @@ class Meta(AssetSerializer.Meta):
                   'downloads',
                   )
 
+    def get_permissions(self, obj):
+        try:
+            # TODO refactored this code if possible.
+            # Quite redundant with `ObjectPermissionHelper.get_assignments_queryset`
+            assignments = self.context[
+                'object_permissions_per_object'].get(obj.pk)
+            context = self.context
+            request = self.context.get('request')
+            user_assignments = []
+            filtered_user_ids = None
+            user = request.user
+
+            if not user or user.is_anonymous():
+                filtered_user_ids = [obj.owner_id]
+            elif not ObjectPermissionHelper.user_can_share(obj, user):
+                # Display only users' permissions if they are not allowed to modify
+                # others' permissions
+                filtered_user_ids = [obj.owner_id, user.pk]
+
+            for assignment in assignments:
+                if (filtered_user_ids is None or
+                        assignment.user_id in filtered_user_ids):
+                    user_assignments.append(assignment)
+
+            context.update({'asset': obj})
+            context.update({'asset_uid': obj.uid})
```

Comment:
A quick glance at DRF makes me think `context` is a `dict`; could we `context[key] = value` instead of calling `update()`, or is `context` actually something fancier?